Marlin/Marlin_main.cpp

```diff
@@ -1659,19 +1659,19 @@ static void clean_up_after_endstop_or_probe_move() {
 
       // If Z needs to raise, do it before moving XY
       if (current_position[Z_AXIS] < z) {
-        feedrate = homing_feedrate[Z_AXIS];
+        feedrate = (feed_rate != 0.0) ? feed_rate : homing_feedrate[Z_AXIS];
         current_position[Z_AXIS] = z;
         line_to_current_position();
       }
 
-      feedrate = XY_PROBE_FEEDRATE;
+      feedrate = (feed_rate != 0.0) ? feed_rate : XY_PROBE_FEEDRATE;
       current_position[X_AXIS] = x;
       current_position[Y_AXIS] = y;
       line_to_current_position();
 
       // If Z needs to lower, do it after moving XY
       if (current_position[Z_AXIS] > z) {
-        feedrate = homing_feedrate[Z_AXIS];
+        feedrate = (feed_rate != 0.0) ? feed_rate : homing_feedrate[Z_AXIS];
         current_position[Z_AXIS] = z;
         line_to_current_position();
       }
```

Comment:
Or French - if you want to emigrate to Canada - when xxx becomes president.

Let's return to the code.
I hoped you'd have been convinced when you insisted in the current `do_probe_raise()`. Will take some while to find out what it should look like. My feeling still says that `z_raise` is enough - but i have to look up all occurrences.
    
We have to many 'unintelligent' moves and 'pseudonymes'.
And we need more of then to differentiate between 'software endstop clipped' and unconstrained ones.   

The big differences in `run_z_probe()`  between DELTA and !DELTA make me angry.